### PR TITLE
[KYUUBI #6562] Support zorder clause syntax to use brackets

### LINKED ISSUE #6562: [Improvement] Zorder clause syntax support brackets
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, the zorder clause syntax does not support brackets.

### How should we improve?

Modify `KyuubiSparkSQL.g4` syntax to support

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by huangxiaopingRD**
# :mag: Description
## Issue References 🔗

This pull request fixes #6562

## Describe Your Solution 🔧

- Modify `KyuubiSparkSQL.g4` to support the zorder clause syntax to use brackets

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:
zorder clause syntax can't use brackets

#### Behavior With This Pull Request :tada:
zorder clause syntax can use brackets

#### Related Unit Tests

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala [hunk 1]
```diff
@@ -746,14 +746,25 @@ trait ZorderSuiteBase extends KyuubiSparkSQLExtensionTest with ExpressionEvalHel
                 UnresolvedFunction("current_date", Seq.empty, false)),
               UnresolvedRelation(TableIdentifier("p")))))))
 
-    // TODO: add following case support
-    intercept[ParseException] {
-      parser.parsePlan("OPTIMIZE p zorder by (c1)")
```
Comment (pan3793) [inline extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: Does Spark SQL support `ORDER BY` with parentheses

Comment (huangxiaopingRD) [reply extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: No, Spark does not support it.

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: then I think we don't need to support that too

Comment (pan3793) [reply extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: I think we can delete those code and comments, unless @ulysses-you and @cfmcgrady have other ideas

Comment (ulysses-you) [reply extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: Seems added by @cfmcgrady , I'm fine to just remove the comment

Comment (cfmcgrady) [reply extensions/spark/kyuubi-extension-spark-3-3/src/test/scala/org/apache/spark/sql/ZorderSuiteBase.scala:751]: Just following the deltalake. https://docs.delta.io/latest/optimizations-oss.html#z-ordering-multi-dimensional-clustering

**Pull request closed without merging**